tools/AWSFeedStorage.py

```diff
@@ -17,22 +19,63 @@ def __init__(self, uri):
 
         self.logger = logging.getLogger(__name__)
 
+        u = urlparse(uri)
+        self.s3_file = u.path[1:]
         self.s3_bucket = settings['AWS_S3_BUCKET']
-        self.s3_file = settings['AWS_S3_FILE_NAME']
-        self.s3 = boto3.resource('s3')
+        self.s3 = boto3.client('s3')
         self.dynamodb = DynamoDBConnector()
 
+    def _generate_datetag(self):
+        curdate = datetime.now()
+        return "{}".format(curdate.strftime('%Y%m%d'))
+
+    def _get_last_result(self):
+        try:
+            objs = self.s3.list_objects_v2(
+                Bucket=self.s3_bucket,
+                Prefix=self.s3_file
+            ).get('Contents', [])
+        except ClientError:
+            self.logger.warning('Could not connect to s3 bucket.')
+            return ''
+
+        if not objs:
+            self.logger.warning('Could not get last result file.')
+            return ''
+
+        last_added = [obj['Key'] for obj in sorted(
+            objs,
+            key=lambda obj: obj['LastModified']
+        )][0]
```

Comment:
also what happens when there are no files?

Reply:
AWS sends a response without the "Contents" key, so `objs` is empty, and this case is catched earlier in the code to log an error and return an empty string.